api/src/decorators/check-authorization.decorator.ts

```diff
@@ -0,0 +1,21 @@
+import { SetMetadata } from '@nestjs/common';
+
+/**
+ * @description Decorator to inject a IS_PUBLIC_KEY metadata to the handler, which will be read by the JwtAuthGuard to allow public access to the handler.
+ */
+
+export const AUTHORIZATION_CHECKS = 'AUTHORIZATION_CHECKS';
+
+export const AUTHORIZATION_STRATEGIES = {
+  USER_ID: 'userId',
+  ROLE: 'role',
+} as const;
+
+export type AuthorizationCheck = [
+  (typeof AUTHORIZATION_STRATEGIES)[keyof typeof AUTHORIZATION_STRATEGIES],
+  params?: unknown[],
+];
+
+export const CheckAuthorization = (
```

Comment:
Naming is quite confusing. Authorization is Checked already checked by the time this decorator is executed